### ISSUE TITLE
exception generated when cannot autolaunch D-Bus without X11 $DISPLAY

### ISSUE THREAD

**Issue by abraunegg** [abraunegg/onedrive#365]
If running a version with compiled in notification support, and running in a terminal via SSH (despite X11 being available on the system and active), an exception is generated:
```
[alex@centos7full onedrive-pr355]$ ./onedrive --confdir '~/.config/onedrive-personal' --monitor --verbose --verbose
[DEBUG] homePath: HOME environment variable set
[DEBUG] homePath: /home/alex
[DEBUG] configDirBase: WARNING - no XDG_CONFIG_HOME environment variable set
[DEBUG] configDirBase: /home/alex/.config
[DEBUG] configDirName: CLI override to set configDirName to: ~/.config/onedrive-personal
[DEBUG] configDirName: A '~' was found in configDirName, using the calculated 'homePath' to replace '~'
Loading config ...
Using Config Dir: /home/alex/.config/onedrive-personal
[DEBUG] sync_dir: Getting syncDir from config value sync_dir
[DEBUG] sync_dir: A '~' was found in configured sync_dir, automatically expanding as SHELL and USER environment variable is set
[DEBUG] syncDir: /home/alex/OneDrivePersonal
Initializing the OneDrive API ...
Opening the item database ...
All operations will be performed in: /home/alex/OneDrivePersonal
Got exception from showing notification: dnotify.NotificationError@src/notifications/dnotify.d(50): Cannot autolaunch D-Bus without X11 $DISPLAY
----------------
src/notifications/dnotify.d:170 void dnotify.Notification.show() [0x5e6d8c]
src/log.d:119 void log.notify!(immutable(char)[]).notify(immutable(char)[]) [0x5aa182]
src/log.d:58 void log.logAndNotify!(immutable(char)[]).logAndNotify(immutable(char)[]) [0x5aa0a3]
src/main.d:395 _Dmain [0x5afc63]
Initializing the Synchronization Engine ...
Account Type: personal
Default Drive ID: <redacted>
Default Root ID: <redacted>
Remaining Free Space: 5368709120
Fetching details for OneDrive Root
OneDrive Root exists in the database
Trying to restore the upload session ...
The file does not exist anymore
Got exception from showing notification: dnotify.NotificationError@src/notifications/dnotify.d(50): Cannot autolaunch D-Bus without X11 $DISPLAY
----------------
src/notifications/dnotify.d:170 void dnotify.Notification.show() [0x5e6d8c]
src/log.d:119 void log.notify!(immutable(char)[]).notify(immutable(char)[]) [0x5aa182]
src/log.d:58 void log.logAndNotify!(immutable(char)[]).logAndNotify(immutable(char)[]) [0x5aa0a3]
src/main.d:488 _Dmain [0x5b020e]
Initializing monitor ...
OneDrive monitor interval (seconds): 45
Monitor directory: .
^CGot termination signal, shutting down db connection
```

No such exception occurs if application started via Terminal Session inside X11 session.

**Comment by norbusan** [comment]
Ah, indeed. Thanks for the report. I will fix it soon!

**Comment by abraunegg** [comment]
It was just a stock standard centos 7 minimal install - if you cant reproduce no worries - will see what I can come up with or even provide a ova of the test system i am using

**Comment by norbusan** [comment]
Ok, I try a docker image. I have already some code written, but cannot test it ;-)

**Comment by norbusan** [comment]
@abraunegg see #367

**Comment by abraunegg** [comment]
Resolved by PR #367
